infra/modules/providers/azure/app-service/main.tf

```diff
@@ -43,23 +55,23 @@ resource "azurerm_app_service" "appsvc" {
 }
 
 resource "null_resource" "acr_webhook_creation" {
-  count      = var.docker_enable_ci == true && var.azure_container_registry_name != "" ? length(keys(var.app_service_name)) : 0
+  count      = var.docker_enable_ci == true && var.azure_container_registry_name != "" ? length(keys(var.app_service_config)) : 0
   depends_on = [azurerm_app_service.appsvc]
 
   triggers = {
-    images_to_deploy = "${join(",", values(var.app_service_name))}"
+    images_to_deploy = "${join(",", [for k, v in var.app_service_config : v.image])}"
     acr_name         = var.azure_container_registry_name
   }
 
   provisioner "local-exec" {
-    command = "az acr webhook create --registry ${var.azure_container_registry_name} --name ${replace(lower(element(keys(var.app_service_name), count.index)), "-", "")}${replace(lower(terraform.workspace), "-", "")}${local.acr_webhook_name} --actions push --uri $(az webapp deployment container show-cd-url -n ${lower(element(keys(var.app_service_name), count.index))}-${lower(terraform.workspace)} -g ${data.azurerm_resource_group.appsvc.name} --query CI_CD_URL -o tsv)"
+    command = format("az acr webhook create --registry %s --name %s%s%s --actions push --uri $(az webapp deployment container show-cd-url -n %s-%s -g %s --query CI_CD_URL -o tsv)", var.azure_container_registry_name, replace(lower(element(keys(var.app_service_config), count.index)), "-", ""), replace(lower(terraform.workspace), "-", ""), local.acr_webhook_name, lower(element(keys(var.app_service_config), count.index)), lower(terraform.workspace), data.azurerm_resource_group.appsvc.name)
```

Comment:
Can this be broken onto multiple lines to improve readability? Unsure if terraform allows this.

Also consider this approach as I found it to improve readability by adding names to the string substitutions:
https://www.terraform.io/docs/provisioners/local-exec.html#environment
```hcl
resource "aws_instance" "web" {
  # ...

  provisioner "local-exec" {
    command = "echo $FOO $BAR $BAZ >> env_vars.txt"

    environment = {
      FOO = "bar"
      BAR = 1
      BAZ = "true"
    }
  }
}
```

Reply:
Yeah, I'm so glad you're pointing this out too. Yeah, I'm going to look for a better alternative. Thanks for giving resources to start with!

Reply:
Making this change was interesting because I ran into errors trying to properly place the string functions (i.e. replace(lower(local.app_names[count.index]), "-", "")).  I opted to place all the sting functions within the environment block. I think this is a situation where one should one place for all string functions. Choose either the 'command' or the 'environment' block but not both because that's prone to errors.

```hcl
  # ...
 provisioner "local-exec" {
    command = "az acr webhook create --registry $ACRNAME --name $APPNAME$WRKSPACE$WEBHOOKNAME . . ."
    
    environment = {
      ACRNAME      = var.azure_container_registry_name
      APPNAME      = replace(lower(local.app_names[count.index]), "-", "")
   }
  }
```